src/uv-common.c

```diff
@@ -209,6 +209,9 @@ int uv_ip6_addr(const char* ip, int port, struct sockaddr_in6* addr) {
     addr->sin6_scope_id = atoi(zone_index);
 #else
     addr->sin6_scope_id = if_nametoindex(zone_index);
+
+    if (addr->sin6_scope_id == 0)
+      return -errno;
```

Comment:
I think it depends if we consider it a potential security liability or not, as mentioned in #1313.